src/components/course/data/utils.jsx

```diff
@@ -149,74 +149,199 @@ export function findCouponCodeForCourse(couponCodes, catalogList = []) {
+ * @param {object} args.secondOffer An enterprise offer to compare with the first offer.
+ *
+ * @returns The sort comparison value.
+ */
+export const compareRemainingBalanceForUser = ({ firstOffer, secondOffer }) => {
+  const firstOfferRemainingBalanceForUser = firstOffer.remainingBalanceForUser;
+  const secondOfferRemainingBalanceForUser = secondOffer.remainingBalanceForUser;
+
+  if (firstOfferRemainingBalanceForUser && secondOfferRemainingBalanceForUser) {
+    if (firstOfferRemainingBalanceForUser < secondOfferRemainingBalanceForUser) {
+      return -1;
+    }
+    if (firstOfferRemainingBalanceForUser >= secondOfferRemainingBalanceForUser) {
+      return 1;
+    }
+  }
+  if (!firstOfferRemainingBalanceForUser && secondOfferRemainingBalanceForUser) {
+    return -1;
+  }
+  if (firstOfferRemainingBalanceForUser && !secondOfferRemainingBalanceForUser) {
+    return 1;
+  }
+  return 0;
+};
+
+/**
+ * Determines the sort order of two enterprise offers based on the
+ * remaining balance for the offer. Returns -1 if the first offer should
+ * be prioritized, 1 if the second offer should be prioritized, and 0 if
+ * there is no difference in priority.
+ *
+ * @param {object} args
+ * @param {object} args.firstOffer An enterprise offer to compare with the second offer.
+ * @param {object} args.secondOffer An enterprise offer to compare with the first offer.
+ *
+ * @returns The sort comparison value.
+ */
+export const compareRemainingBalance = ({ firstOffer, secondOffer }) => {
+  const firstOfferRemainingBalance = firstOffer.remainingBalance;
+  const secondOfferRemainingBalance = secondOffer.remainingBalance;
+  if (firstOfferRemainingBalance && secondOfferRemainingBalance) {
+    if (firstOfferRemainingBalance < secondOfferRemainingBalance) {
+      return -1;
+    }
+    if (firstOfferRemainingBalance >= secondOfferRemainingBalance) {
+      return 1;
+    }
+  }
+  if (!firstOfferRemainingBalance && secondOfferRemainingBalance) {
+    return -1;
+  }
+  if (firstOfferRemainingBalance && !secondOfferRemainingBalance) {
+    return 1;
+  }
+  return 0;
+};
+
+/**
+ * Determines the sort order of two enterprise offers based on the
+ * remaining applications for the offer. Returns -1 if the first offer should
+ * be prioritized, 1 if the second offer should be prioritized, and 0 if
+ * there is no difference in priority.
+ *
+ * @param {object} args
+ * @param {object} args.firstOffer An enterprise offer to compare with the second offer.
+ * @param {object} args.secondOffer An enterprise offer to compare with the first offer.
+ *
+ * @returns The sort comparison value.
+ */
+export const compareRemainingApplications = ({ firstOffer, secondOffer }) => {
+  const firstOfferRemainingApplications = firstOffer.remainingApplications;
+  const secondOfferRemainingApplications = secondOffer.remainingApplications;
+  if (firstOfferRemainingApplications && secondOfferRemainingApplications) {
```

Comment:
Perhaps this wouldn't simplify things, but I wonder if some of these `if` statements could instead be solved in a helper function? Something like this:
```
const compareOffers = (firstOffer, secondOffer, property) => {
   const firstOfferValue = firstOffer[property];
   const secondOfferValue = secondOffer[property];
   if (firstOfferValue && !secondOfferValue) return -1;
   if (!firstOfferValue && secondOfferValue) return 1;
   if (firstOfferValue < secondOfferValue) return -1;
   if (firstOfferValue >= secondOfferValue) return 1;
   return 0;
}
```
Feel free to ignore if it doesn't add any value! :)

Reply:
Actually looks like you already did this!